### PR TITLE
Add sources for some strategies.

### PR DESCRIPTION
**Pull request by drvinceknight**
Ongoing work on #225

### CODE REVIEW

File: axelrod/strategies/calculator.py [hunk 1]
```diff
@@ -10,6 +10,11 @@ class Calculator(Player):
     """
     Plays like (Hard) Joss for the first 20 rounds. If periodic behavior is
     detected, defect forever. Otherwise play TFT.
+
+
+    Names:
+
+    - Calculator: Original name by Marc Harper
```
Comment (marcharper) [inline axelrod/strategies/calculator.py]: It's not mine -- it's in PRISON and likely predates it (it's in the file of classic strategies). Maybe one of the original Axelrod tournaments?

File: axelrod/strategies/alternator.py [hunk 2]
```diff
@@ -5,7 +5,13 @@
 
 
 class Alternator(Player):
-    """A player who alternates between cooperating and defecting."""
+    """
+    A player who alternates between cooperating and defecting.
+
+    Names
+
+    - Alternator: [Bendor1993]_
```
Comment (marcharper) [inline axelrod/strategies/alternator.py]: Surely there is an earlier description ...

**Comment by marcharper** [comment]
We need to decide if any reference will do or if we are going to try to track down the original reference in all cases.

File: axelrod/strategies/alternator.py (hunk 2 above)
Comment (drvinceknight) [inline axelrod/strategies/alternator.py]: I thought so too but I struggled to find it.

File: axelrod/strategies/calculator.py (hunk 1 above)
Comment (drvinceknight) [inline axelrod/strategies/calculator.py]: Thanks. I don't think it's one of the original tournaments I'll go with PRISON.

**Comment by drvinceknight** [comment]
> We need to decide if any reference will do or if we are going to try to track down the original reference in all cases.
> Merge state

What we've been doing so far is any description of the name in the literature. This will do for now and is an improvement over nothing.

File: axelrod/strategies/alternator.py (hunk 2 above)
Comment (drvinceknight) [inline axelrod/strategies/alternator.py]: I'll put Axelrod's book in (it's what I did for Cooperator and Alternator). :+1:

**Comment by drvinceknight** [comment]
I've just added a few more. I'm missing a few of the ZD ones in `axelrod/strategies/memory_one.py`, would you be able to add those perhaps @marcharper? I'm going to probably finish off everything else by the end of today (nearly there now).

**Comment by drvinceknight** [comment]
> would you be able to add those perhaps @marcharper?

If you have the time please push directly to this branch but no worries if not, I can circle back and hunt them down :) :+1:

**Comment by drvinceknight** [comment]
After 383c488 there are two remaining strategies:

- Soft Joss. Soft Joss is GTFT with a different parameter. We need to
find the literature with that particular parameter. Note that I've added
a source [Torino2014] that lists SoftJoss as another name for GTFT and
vice versa.
- I cannot find EXTORT-4 (I've looked here:  https://plato.stanford.edu/entries/prisoner-dilemma/)

FYI, the way I've gone about this PR is for any given strategy that was tricky:

- google searching anything/everything related to the name
- git blaming to have a guess at who implemented it
- looking through git history of a file
- reading papers that I think might be relevant.

**Comment by marcharper** [comment]
I'm trying to track it down. I suspect it's one with \Chi = 4 but I haven't found it yet. It's possible that I made this one up.

I have found some other concrete ZD examples in case we want to add more examples from the literature:
* (11/13, 1/2, 7/26, 0) from Press and Dyson
* ZDmischief (0.8, 0.6, 0.1, 0) an ZDextortion (0.64, 0.18, 0.28, 0) from [this paper](https://arxiv.org/pdf/1308.2576.pdf)
 There's a memory-two generalization in [this paper](http://math.uchicago.edu/~may/REU2014/REUPapers/Li,Siwei.pdf) on page 21 , as well as the memory-one (15/16, 1/2, 7/16, 1/16) 
* Looks like maybe one or two more in [this paper](https://pdfs.semanticscholar.org/824a/2123e1de5aa2e971fa9b1bf167b8ff246aa5.pdf) PZDR (1.0, 0.35, 0.75, 0.1) (but looks like donation game matrix)
* Some in [this paper](http://web.evolbio.mpg.de/~hilbe/Research_files/Hilbe%20et%20al%20(GEB%202015)%20Partners%20or%20rivals.pdf), see the caption for Fig 3

**Comment by marcharper** [comment]
Here's a [source](http://euler.fd.cvut.cz/predmety/game_theory/lecture_repeat.pdf) for "SOFT JOSS" (lecture notes). It's also implemented in PRISON.

**Comment by drvinceknight** [comment]
> Here's a source for "SOFT JOSS" (lecture notes). It's also implemented in PRISON.

I wasn't quite satisfied with those lecture notes when I found them for some reason (not a valid one): have gone with Prison!

> I suspect it's one with \Chi = 4 but I haven't found it yet. It's possible that I made this one up.

I think it is probably yours, I've gone with that anyway for now. We can always address at a later date :)

> I have found some other concrete ZD examples

I'll copy those to #379 :)

If I'm not mistaken I think 9376619 now completes all the sources and closes #225 (assuming we're happy with the sources I've put in).

File: axelrod/strategies/apavlov.py [hunk 3]
```diff
@@ -73,13 +74,14 @@ def reset(self):
...
-    closely as can be determined.
-
     APavlov attempts to classify its opponent as one of four strategies:
     Cooperative, ALLD, STFT, or Random. APavlov then responds in a manner
     intended to achieve mutual cooperation or to defect against
     uncooperative opponents.
+
+    Names:
+
+    - Adaptive Pavlov 2007: [Li2011]_
```
Comment (marcharper) [inline axelrod/strategies/apavlov.py]: 2011?

File: axelrod/strategies/hunter.py [hunk 4]
```diff
@@ -8,7 +8,12 @@
 
 
 class DefectorHunter(Player):
-    """A player who hunts for defectors."""
+    """A player who hunts for defectors.
+
+    Names:
+
+    Defector Hunter: Original name by Marc Harper
```
Comment (marcharper) [inline axelrod/strategies/hunter.py]: These hunters are Karol I think except for cycle hunter and eventual cycle hunter

File: axelrod/strategies/meta.py [hunk 5]
```diff
@@ -182,7 +203,12 @@ def meta_strategy(self, results, opponent):
 
 
 class MetaHunter(MetaPlayer):
-    """A player who uses a selection of hunters."""
+    """A player who uses a selection of hunters.
+
+    Names
+
+    - Meta Hunter: Original name by Marc Harper
```
Comment (marcharper) [inline axelrod/strategies/meta.py]: Karol? (Not me)

File: axelrod/strategies/titfortat.py [hunk 6]
```diff
@@ -498,6 +498,9 @@ class SlowTitForTwoTats(Player):
     A player plays C twice, then if the opponent plays the same move twice,
     plays that move.
 
+    Names:
+
+    - Slow tit for two tats: Original name by Ranjini Das
```
Comment (marcharper) [inline axelrod/strategies/titfortat.py]: PRISON

**Comment by drvinceknight** [comment]
I think I've caught all of those now @marcharper. Thanks for picking them up!

File: axelrod/strategies/meta.py [hunk 7]
```diff
@@ -225,7 +251,12 @@ def meta_strategy(results, opponent):
 
 
 class MetaHunterAggressive(MetaPlayer):
-    """A player who uses a selection of hunters."""
+    """A player who uses a selection of hunters.
+
+    Names
+
+    - Meta Hunter: Original name by Marc Harper
```
Comment (marcharper) [inline axelrod/strategies/meta.py]: Meta Hunter Aggressive

Comment (drvinceknight) [inline axelrod/strategies/meta.py]: Good catch. 9af08c2

**Comment by drvinceknight** [comment]
Two very minor fixes in: ee77f4a 09b5994

**Comment by drvinceknight** [comment]
FYI Have added a couple more very little fixes (just minor details like year etc that I found whilst getting the bibtex for the moran paper).

**Comment by meatballs** [comment]
Is this finished and ready to be merged?

**Comment by drvinceknight** [comment]
Yes, assuming you're ok with it.

**Pull request merged**